torchtune/training/metric_logging.py

```
@@ -48,6 +48,17 @@ def save_config(config: DictConfig) -> Path:
         log.warning(f"Error saving config.\nError: \n{e}.")
 
 
+def flatten_dict(d, parent_key="", sep="."):
```

Please add a small comment here explaining the expected input format and output format. Also kwargs only after `d` param.

---

Added a proper docstring to align to `save_config` - let me know if it's not clear!

> Also kwargs only after d param.

I wasn't exactly sure what this meant. Did you mean to explicitly use `**kwargs` then parse the dict in the function?

For now I placed `sep` before `parent_key` since `sep` is more likely to be changed by a user (although setting `parent_key` may be useful if we ever wanted to prefix the config params with `torchtune_config` or similar)

---

Ahh, sorry I wasn't being clear. In general, for any optional params we like to use the wildcard operator to force the user to specify the param using keyword only. An example is like so:

```python
def foo(a: int, *, b: Optional[int] = None) -> int:
   ...
```

This forces the user to refer to the parameter `b` in function `foo` by name:

```python
foo(1, b=2)
```

Not only does this improve readability of the code, reduce the possibility for silent errors by sending parameters in with the wrong ordering, it also makes it easier to deprecate any kwargs that we might want to change down the line. So in this case, I'd like to see:

```python
def flatten_dict(d: Dict[str, Any], *, parent_key: str = "", sep: str = ".") -> Dict[str, Any]:
    ...
```

Does that make sense?

---

Ahh gotcha, love it, done.